gobblin-core/src/main/java/gobblin/writer/BaseDataWriter.java

```diff
@@ -0,0 +1,76 @@
+/* (c) 2015 NerdWallet All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.writer;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.configuration.State;
+import gobblin.util.ForkOperatorUtils;
+import gobblin.util.HadoopUtils;
+import gobblin.util.WriterUtils;
+
+import java.io.IOException;
+import java.net.URI;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.fs.FileSystem;
+import org.apache.hadoop.fs.Path;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * An implementation of {@link DataWriter} does the work of setting the output/staging dir
+ * and creating the FileSystem instance.
+ *
+ * @author akshay@nerdwallet.com
+ */
+public abstract class BaseDataWriter<D> implements DataWriter<D> {
```

Comment:
Actually, I believe it will only work with an instance of `org.apache.hadoop.fs.FileSystem`, so its only HDFS right? If this is the case, I'll change it to `HdfsDataWriter`. Otherwise, I'll use `FsDataWriter`.